src/block/tile/Sign.php

```diff
@@ -111,5 +151,8 @@ public function setEditorEntityRuntimeId(?int $editorEntityRuntimeId) : void{
 
 	protected function addAdditionalSpawnData(CompoundTag $nbt) : void{
 		$nbt->setString(self::TAG_TEXT_BLOB, implode("\n", $this->text->getLines()));
+		$nbt->setString(self::TAG_TEXT_COLOR, $this->textColor->name());
```

Comment:
name() should not be used for this. The internal PocketMine colour names are not guaranteed to match the MC internals. This may break without warning.